dlt/destinations/impl/databricks/configuration.py

```diff
@@ -0,0 +1,112 @@
+from typing import ClassVar, Final, Optional, Any, Dict, List
+
+from dlt.common.typing import TSecretStrValue
+from dlt.common.configuration.exceptions import ConfigurationValueError
+from dlt.common.configuration.specs.base_configuration import CredentialsConfiguration, configspec
+from dlt.common.destination.reference import DestinationClientDwhWithStagingConfiguration
+
+
+CATALOG_KEY_IN_SESSION_PROPERTIES = "databricks.catalog"
+
+
+@configspec
+class DatabricksCredentials(CredentialsConfiguration):
+    catalog: Optional[str] = None
+    server_hostname: str = None
+    http_path: str = None
+    access_token: Optional[TSecretStrValue] = None
+    client_id: Optional[str] = None
+    client_secret: Optional[TSecretStrValue] = None
+    session_properties: Optional[Dict[str, Any]] = None
+    connection_parameters: Optional[Dict[str, Any]] = None
+    auth_type: Optional[str] = None
+
+    connect_retries: int = 1
+    connect_timeout: Optional[int] = None
+    retry_all: bool = False
+
+    _credentials_provider: Optional[Dict[str, Any]] = None
+
+    __config_gen_annotations__: ClassVar[List[str]] = [
+        "server_hostname",
+        "http_path",
+        "catalog",
+        "schema",
+    ]
+
+    def __post_init__(self) -> None:
```

Comment:
we also have the on_resolved function that gets called after the configuration is resolved, maybe you can use that, or do you need this to run before?

Reply:
Yeah on_resolved should work. Didn't touch these classes much from the previous PR since it seemed to work, needs some tweaks.

Reply:
For now I removed a lot of this and am only supporting the "personal access token" auth.   
I don't have a way to test oauth atm, will need to be done somehow so we're not prompting on every connection.